<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8' />
    <meta http-equiv="X-UA-Compatible" content="chrome=1" />
    <meta name="description" content="Hugh Cross' personal github page" />

    <!-- CSS -->
   <!--  <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css"> -->
    <link rel="stylesheet" href="stylesheets/poole.css">
    <!-- <link rel="stylesheet" href="stylesheets/syntax.css"> -->
    <link rel="stylesheet" href="stylesheets/hyde.css">
    <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=PT+Sans%3A400,400italic,700%7CAbril+Fatface">


    <title>Hugh Cross' Personal GitHub Page</title>
  </head>

  <body class="theme-base-0G"> -->

    <!-- HEADER 
    <div id="header_wrap" class="outer">
        <header class="inner">
          <a id="forkme_banner" href="https://github.com/hughcross">View on GitHub</a>

          <h1 id="project_title">Hugh Cross' GitHub Page</h1>
          <h2 id="project_tagline">Bioinformatic Tools for Biodiversity and Evolutionary Genomics</h2>

         
        </header>
    </div> -->

    <!-- SIDEBAR -->
        <div class="sidebar">
  <!--
  <div class="container sidebar-sticky">
  -->
  <div class="container">
    <div class="sidebar-about">
      <!--
      <h1>
        <a href="/">
          Biopython
        </a>
      </h1> -->
      
      <div class="sidebar-logo">
        <a href="/">
          <img src="images/hughcross_profile_photo.jpg" alt="hcross Logo" title="hcross_forest_photo" width="100%"/>
        </a>
      </div> 
      <p class="lead"></p>
    </div>

    <nav class="sidebar-nav">
      <!--
      <a class="sidebar-nav-item" href="/">Home</a>
      -->
            <h3 class="sidebar-nav-item" ><em>Contents:</em></h3>
            <a class="sidebar-nav-item" href="#teaching_resources">Teaching Resources</a>
            <a class="sidebar-nav-item" href="links.html" target="_blank" rel="noopener noreferrer">Jupyter Notebooks</a>
            <a class="sidebar-nav-item" href="#basic_tools">Basic Tools</a>
            <a class="sidebar-nav-item" href="#ncbi_tools">NCBI Tools</a>
            
            <br/>
            
            <h3 class="sidebar-nav-item" ><em>Repositories:</em></h3>

            <a class="sidebar-nav-item" href="https://github.com/hughcross/seq_tools.git" target="_blank" rel="noopener noreferrer">seq_tools</a>
            <a class="sidebar-nav-item" href="https://github.com/hughcross/genbanking" target="_blank" rel="noopener noreferrer">genbanking</a>
            <br/>

            <h3 class="sidebar-nav-item" ><em>About:</em></h3>

            <a class="sidebar-nav-item" href="https://www.researchgate.net/profile/Hugh-Cross/research" target="_blank" rel="noopener noreferrer">Publications</a>
            
            <br/>

       
            <!-- <h3 class="sidebar-nav-item" ><em>Other Useful Links:</em></h3>

            <a class="sidebar-nav-item" href="https://github.com/hughcross">GitHub home page</a>
            <a class="sidebar-nav-item" href="http://biopython.org/wiki/Biopython">Biopython main wiki</a> -->
            
      <!--
      The GitHub snapshots for each tag  misses things like compiled Tutorial.pdf
      <a class="sidebar-nav-item" href="https://github.com/biopython/biopython/archive/v1.68.zip">Download</a>
      -->
      
    </nav>
    
  </div>
  <!-- <div class="sidebar-footer">
      Biopython version 1.68<br>
      &copy; 2017. All rights reserved.
  </div> -->
</div>


    <!-- MAIN CONTENT : new main content, shifting to other css rules
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner"> -->


          <div class="content container">
      <a href="https://github.com/hughcross" class="head">View on GitHub</a>
      <hr class="head"/>
      <div class="page">
  <!-- <h1 class="page-title">Biopython</h1>
  <p>See also our
<a href="http://news.open-bio.org/category/obf-projects/biopython/feed/rss">News feed</a>
and <a href="https://twitter.com/biopython">Twitter</a>.</p> -->

<!-- <h3 id="introduction">Introduction</h3>  -->

          <h1 id="project_title">Hugh Cross' GitHub Page</h1>
          <h2 id="project_tagline">Bioinformatic Tools for Biodiversity and Evolutionary Genomics</h2>
          <br/>

        <!-- <h1>Forest Health Genomics</h1> -->
        <p>This is my personal GitHub page. I am a bioinformatician in the  <a href="http://www.otago.ac.nz/anatomy/index.html" target="_blank" rel="noopener noreferrer">Department of Anatomy</a> at the <a href="http://www.otago.ac.nz/" target="_blank" rel="noopener noreferrer">University of Otago</a> in Dunedin, New Zealand. This site will be the central place to link to my GitHub projects and repositories. I have added some basic (but hopefully useful) scripts, but will populate this site with more content as they are developed. </p>
         <br/>

         <h2 id='teaching_resources'>Teaching Resources</h2>

         <p>I have been running a bioinformatics help session in the Anatomy department tea room. This was postponed during the pandemic but will be starting again soon. Most of the material from these sessions is available on the <a href="https://otagomohio.github.io/hackyhour/" target="_blank" rel="noopener noreferrer">hacky hour webpage</a>. These sessions range from <a href="https://otagomohio.github.io/hackyhour/sessions/2019_09_11.html" target="_blank" rel="noopener noreferrer">getting started writing scripts</a> to some <a href="https://otagomohio.github.io/hackyhour/sessions/2019_08_14.html" target="_blank" rel="noopener noreferrer">guidelines on troubleshooting command line errors</a>. More material will be coming soon. </p>

         <p>As part of the Otago Carpentries group (<a href="https://otagocarpentries.github.io/" target="_blank" rel="noopener noreferrer"><b>main link</b></a>)  I have helped put on frequent workshops and training sessions. <a href="https://otagocarpentries.github.io/past" target="_blank" rel="noopener noreferrer">Here is a link to past workshops</a>, which  contain a great deal of content. The <a href="https://otagocarpentries.github.io/bioinformatics-spring-school-2020/" target="_blank" rel="noopener noreferrer"><b>Bioinformatics Spring School</b></a>, held in November, 2020, provides a framework for future large events through Otago Carpentries. It consisted of both basic lessons and specific training in several areas, such as gene expression and population genetics.  </p>

         <p>I have taught several workshops on environmental DNA (eDNA), and organised an <a href="https://otagoedna.github.io/eDNA_Workshop_Nov_2019/" target="_blank" rel="noopener noreferrer">eDNA conference</a> attended by researchers from throughout New Zealand and the Pacific region. As part of the conference, I taught a <a href="https://otagoedna.github.io/2019_11_28_edna_course/" target="_blank" rel="noopener noreferrer">one-day workshop on metabarcoding</a>, which included an optional day for beginners to learn the basics of the command line and R. This was organised with the help of Otago Carpentries. We ran the workshop again as part of the <a href="https://otagocarpentries.github.io/bioinformatics-spring-school-2020/day5.html" target="_blank" rel="noopener noreferrer">Bioinformatics Spring School</a>. </p>

         <p>The growing interest in environmental DNA has led to the creation of an eDNA Hub at the University of Otago. <a href="https://blogs.otago.ac.nz/go/events/" target="_blank" rel="noopener noreferrer">Our inaugural meeting will be held on 25 March</a>. This initiative was organised by Dr. Tina Summerfield (Department of Botany) and myself. The Hub is supported by <a href="https://blogs.otago.ac.nz/go/" target="_blank" rel="noopener noreferrer">Genetics Otago</a> and <a href="https://www.genomics-aotearoa.org.nz/" target="_blank" rel="noopener noreferrer">Genomics Aotearoa</a>. </p>
         

         <br/>

         <h2>Jupyter Notebooks</h2>
         <p>I do most of my Python development using Jupyter notebooks. I have started to export the notebooks in HTML format so I can reference them easily. <a href="links.html" target="_blank" rel="noopener noreferrer">Here is a link</a> to the index page.</p>
        <br/>

        <h2 id='basic_tools'>Some Basic Tools</h2>

        <p>In the repository <a href="https://github.com/hughcross/seq_tools.git" target="_blank" rel="noopener noreferrer">seq_tools</a> I have put a few scripts that I use for manipulating DNA sequence data. A useful one is <b>seq_extractor.py</b>, that I use to extract a subset of sequences from a larger fasta file, from a file with a list of sequence ids. The page contains example files. I have used this script a great deal, as I find I often want to take a subset of sequences from a file (e.g. differentially expressed transcripts from a transcriptome file). I often use it interactively in <a href="http://jupyter.org/" target="_blank" rel="noopener noreferrer">Jupyter Notebook</a>. I have extracted sequences from very large files (e.g. spruce genome, 8 Gb with about 12 million scaffolds). The speed was tremendously improved when I converted the list of sequence ids to a Python set (from hours to minutes). Just goes to show that data structures matter!</p> 

        <p>Another useful tool is <b>revcomp.py</b> that will simply return the reverse complement of either a single sequence, or a fasta file.</p>

        <p>The script <b>seq_name_changer.py</b> is a bit more involved, mostly because it has several options. I like to rename all my sequence names in each file to have more information than the random code that the sequencing machine spits out. I know there are heaps of programs and scripts out there that can change read names in a sequence file, or have that capability among other options. However, I have often found that I want to change the names in a way that doesn't fit with the usual program. This script is a combination of the various name-changing scripts I have written. The option I use often is to conform to RNAseq paired end sequencing format that is required by a few programs. The most common (although not only) format is to have forward reads end in '/1' and reverse reads end in '/2', with nothing after this on the name. This option can be invoked with '-f fwd' or '-f rev' (f for format). Not all programs are fussy about the naming, but some of them will spit out errors, or just stop, without telling you what happened. I find it is good practice to change all my paired end data and stick to this scheme.</p>
        <br/>

        <h2 id='ncbi_tools'>Useful Tools for NCBI files</h2>

        <p>My repository <a href="https://github.com/hughcross/genbanking" target="_blank" rel="noopener noreferrer">genbanking</a> contains a couple of scripts for parsing NCBI files, including BLAST files and Genbank format sequence files. I use these kinds of files so often that I put together some general tools for parsing and extracting information from them. I describe two below, but I will be adding more.</p>

        <p>For running BLAST searches, my preferred output format is tabular (option -outfmt 6). This is because they are human readable for a quick look, but also are tab-delimited so easy to parse. Also, there are many options to customize them. I have written so many scripts that parse one or other formats, based on different needs, that I finally wrote a function that allows any of the fields from a tabular BLAST output to be utilized, either with default or custom tab format. I realize that <a href="http://biopython.org/wiki/Biopython" target="_blank" rel="noopener noreferrer">Biopython</a> has good modules for parsing BLAST files, but only from xml format (last I looked), which is less convenient when you want to examine the files before parsing. My script <b>filter_blast_results.py</b> can filter a BLAST file using up to 22 different parameters, such as minimum alignment length or evalue.</p> 

        <p>I have also written several functions for extracting information from genbank files (.gb). I have written the script <b>genbank_to_fasta.py</b> that uses these functions and the Biopython packages <i>SeqIO</i> and <i>SeqFeature</i> to convert the genbank to a fasta file. Biopython alone can do this, but the sequence names output are just the GI number, which makes it hard to keep track of your sequences in an alignment or tree output (without renaming them). The included script will use the taxon name and either GI or Accession number to name the sequences. There are other options to modify these names, including adding the gene name, or description after the sequence name.</p> 

        <br/>
        
        
        <br/>

        <h2>Additional tools</h2>
        <br/>

        <p>More content coming!</p>
        <br/>

         <p class="footer"><a class="footer" href="https://github.com/poole/hyde" target="_blank" rel="noopener noreferrer">Hyde</a> theme provided by Mark Otto</p>
         <p class='footer'>Published with <a href="http://pages.github.com" target="_blank" rel="noopener noreferrer">GitHub Pages</a></p>



<!-- </section> -->
    </div>
    </div>

    

  </body>
</html>

