<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8' />
    <meta http-equiv="X-UA-Compatible" content="chrome=1" />
    <meta name="description" content="Hugh Cross' personal github page" />

    <!-- CSS -->
   <!--  <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css"> -->
    <link rel="stylesheet" href="stylesheets/poole.css">
    <!-- <link rel="stylesheet" href="stylesheets/syntax.css"> -->
    <link rel="stylesheet" href="stylesheets/hyde.css">
    <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=PT+Sans%3A400,400italic,700%7CAbril+Fatface">


    <title>Hugh Cross' Jupyter Notebooks</title>
  </head>

  <body class="theme-base-0G">
  
      <!-- SIDEBAR -->
        <div class="sidebar">
  <!--
  <div class="container sidebar-sticky">
  -->
  <div class="container">
    <div class="sidebar-about">
      <!--
      <h1>
        <a href="/">
          Biopython
        </a>
      </h1> -->
      
      <div class="sidebar-logo">
        <a href="/">
          <img src="images/180108110357-jupiter-nasa-color-1-exlarge-169.jpg" alt="hcross Logo" title="nasa_jupiter_clouds_photo" width="100%"/>
        </a>
      </div> 
      <p class="lead"></p>
    </div>

    <nav class="sidebar-nav">
      <!--
      <a class="sidebar-nav-item" href="/">Home</a>
      -->
            <h3 class="sidebar-nav-item" ><em>Categories:</em></h3> 
            <a class="sidebar-nav-item" href="#teaching_notebooks">Teaching Notebooks</a>
            <a class="sidebar-nav-item" href="#find_haps">Find Haplotypes</a>
            <a class="sidebar-nav-item" href="#convert_tables">Convert Tables</a>
            <a class="sidebar-nav-item" href="#gff_tools">Tools for GFF</a>
            
            <br/>
            
            <h3 class="sidebar-nav-item" ><em>Handy Links:</em></h3>
            
            <a class="sidebar-nav-item" href="http://jupyter.org/" target="_blank" rel="noopener noreferrer">Jupyter home</a>

            <a class="sidebar-nav-item" href="https://conda.io/docs/" target="_blank" rel="noopener noreferrer">Conda tools</a>
            
            <br/>
       
            <br/>

    </nav>
    
  </div>
</div>

          <div class="content container">
      <a href="https://github.com/hughcross" class="head">View on GitHub</a>
      <hr class="head"/>
      <div class="page">
  
  
            <h1 id="project_title">Jupyter Notebooks</h1>
          <h2 id="project_tagline">Links to some of my notebooks in HTML format</h2>
          <br/>
          <h2 id='teaching_notebooks'>Teaching Notebooks</h2>
          <p>The following notebooks are from a two day environmental DNA (eDNA) workshop I gave in June at the Auckland campus of the University of Otago. It was attended by University students and researchers from across New Zealand. The first half of the first day was team taught by Ngoni Faya (Genomics Aotearoa) and Dinindu Senanayake (NeSI). I prepared and taught the remaining 1.5 days, using material developed by Gert-Jan Jeunen (University of Otago) and myself.  All of these lessons were taught using <a href="https://jupyter.org/hub" target="_blank" rel="noopener noreferrer"><b>Jupyter Hub</b></a> on the <a href="https://www.nesi.org.nz/about-us/background" target="_blank" rel="noopener noreferrer"><b>NeSI server</b></a>, which will now be the model for future workshops through Otago Carpentries. A full schedule and more course material is available on the <a href="https://jupyter.org/hub" target="_blank" rel="noopener noreferrer"><b>course web page</b></a></p> 

          <p> <a href="https://otagoedna.github.io/edna_workshop_june2021/notebook_htmls/01_Metabarcoding_Basics.html" target="_blank" rel="noopener noreferrer">Metabarcoding Basics</a> </p>

          <p> <a href="https://otagoedna.github.io/edna_workshop_june2021/notebook_htmls/02_demultiplex_and_trimming.html" target="_blank" rel="noopener noreferrer">Demultiplexing and Trimming</a> </p>
          
          <p> <a href="https://otagoedna.github.io/edna_workshop_june2021/notebook_htmls/03_denoising_and_clustering.html" target="_blank" rel="noopener noreferrer">Denoising and Clustering</a> </p>
          
          <p> <a href="https://otagoedna.github.io/edna_workshop_june2021/notebook_htmls/04_intro_to_qiime.html" target="_blank" rel="noopener noreferrer">Introduction to Qiime</a> </p>
          
          <p> <a href="https://otagoedna.github.io/edna_workshop_june2021/notebook_htmls/05_taxonomy_assignment.html" target="_blank" rel="noopener noreferrer">Taxonomy Assignment</a> </p>

          <p> <a href="https://otagoedna.github.io/edna_workshop_june2021/notebook_htmls/06_importing_into_R.html" target="_blank" rel="noopener noreferrer">Importing Metabarcoding Data into R</a> </p>

          <p> <a href="https://otagoedna.github.io/edna_workshop_june2021/notebook_htmls/07_getting_started_exploring_data.html" target="_blank" rel="noopener noreferrer">Getting Started Exploring Data</a> </p>

          <p> <a href="https://otagoedna.github.io/edna_workshop_june2021/notebook_htmls/08_graphing_data_with_R.html" target="_blank" rel="noopener noreferrer">Graphing Data with R</a> </p>
          <br/>
          <h2 id='find_haps'>Finding Haplotypes in Alignments</h2>
          <p>The following three notebooks were used to create scripts for a project to determine frequencies of closely related haplotypes in a mock community experiment, in order to evaluate efficacy of denoising/clustering algorithms. The first two also server to illustrate my approach to script development. For larger scripts, my first step is usually to work out the different steps, using one or two example files or samples. Then I often create functions from these initial steps and further troubleshoot these in a subsequent notebook. This makes it easier to reuse any of these steps in other projects, if needed. Finally, if the steps need to be used by a student or other researchers I will export the notebook as a Python script and add Argparse arguments for command line use.</p>
          <p> <a href="https://hughcross.github.io/notebooks/find_haplotypesDEV_pub.html">Find haplotypes</a> </p>
          
          <p> <a href="https://hughcross.github.io/notebooks/find_haps_w_functions.html">Find haplotypes using functions</a>  </p>
          
          <p>The last notebook was used to develop a script for extracting and counting the frequency of haplotypes in the data, using the outputs from the first script. </p>
          <p> <a href="https://hughcross.github.io/notebooks/extractHaplotypesDEL.html">Extract haplotypes from NGS data</a> </p>
          
          <br/>


          
          <h2 id='convert_tables'>Converting Tables</h2>
          <p>Following are a few examples of notebooks for converting table formats. </p>
          <p> <a href="https://hughcross.github.io/notebooks/convert_obitab2qiime.html">Convert Obitools tab to Qiime2</a> </p>
          
          <p> <a href="https://hughcross.github.io/notebooks/Create_genus_table_add_allLandPlants.html">Create sqlite DB from taxonomy table</a> </p>
          
          <p> <a href="https://hughcross.github.io/notebooks/create_trnl_ref_from_blast.html">Create reference sequence from BLAST results</a> </p>
          <br/>
          
          <h2 id='gff_tools'>What the gff</h2>
          <p>Here are a few notebook for converting or modifying annotation files in GFF3 (and some GTF) format. The first one will convert a GTF for use in <a href="https://www.bioinformatics.babraham.ac.uk/projects/seqmonk/">SeqMonk</a> and enable the gene names to be visualised more easily. </p>

          <p> <a href="https://hughcross.github.io/notebooks/convert_gtf_for_seqmnk_tunicateNew.html">Add gene names to GTF for SeqMonk</a> </p>
          
          <p> <a href="https://hughcross.github.io/notebooks/change_values_in_column_of_gff.html">Change values in column of gff</a> </p>
          

          <p> <a href="https://hughcross.github.io/notebooks/convert_range_data_to_gff_with_diff_categories.html">Convert range data to gff with diff categories</a> </p>
          
          
          <p> <a href="https://hughcross.github.io/notebooks/convert_dN-gmap_to_dex-compat_map_gff_DEL.html">Convert dN-gmap to dex-compatible map gff</a> </p>
          
          <p> <a href="https://hughcross.github.io/notebooks/finding_overlapping_ranges_between_two_gffs_dev.html">Finding overlapping ranges between two gff files</a> </p>
          <br/>



           </div>
</div>
          
          
