<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8' />
    <meta http-equiv="X-UA-Compatible" content="chrome=1" />
    <meta name="description" content="Hugh Cross' personal github page" />

    <!-- CSS -->
   <!--  <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css"> -->
    <link rel="stylesheet" href="stylesheets/poole.css">
    <!-- <link rel="stylesheet" href="stylesheets/syntax.css"> -->
    <link rel="stylesheet" href="stylesheets/hyde.css">
    <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=PT+Sans%3A400,400italic,700%7CAbril+Fatface">


    <title>Hugh Cross' Jupyter Notebooks</title>
  </head>

  <body class="theme-base-0G">
  
      <!-- SIDEBAR -->
        <div class="sidebar">
  <!--
  <div class="container sidebar-sticky">
  -->
  <div class="container">
    <div class="sidebar-about">
      <!--
      <h1>
        <a href="/">
          Biopython
        </a>
      </h1> -->
      
      <div class="sidebar-logo">
        <a href="/">
          <img src="images/180108110357-jupiter-nasa-color-1-exlarge-169.jpg" alt="hcross Logo" title="nasa_jupiter_clouds_photo" width="100%"/>
        </a>
      </div> 
      <p class="lead"></p>
    </div>

    <nav class="sidebar-nav">
      <!--
      <a class="sidebar-nav-item" href="/">Home</a>
      -->
            <h3 class="sidebar-nav-item" ><em>Categories:</em></h3> 
            <a class="sidebar-nav-item" href="#teaching_notebooks">Teaching Notebooks</a>
            <a class="sidebar-nav-item" href="#convert_tables">Convert tables</a>
            <a class="sidebar-nav-item" href="#gff_tools">Tools for GFF</a>
            
            <br/>
            
            <h3 class="sidebar-nav-item" ><em>Handy Links:</em></h3>
            
            <a class="sidebar-nav-item" href="http://jupyter.org/" target="_blank" rel="noopener noreferrer">Jupyter home</a>

            <a class="sidebar-nav-item" href="https://conda.io/docs/" target="_blank" rel="noopener noreferrer">Conda tools</a>
            
            <br/>
       
            <br/>

    </nav>
    
  </div>
</div>

          <div class="content container">
      <a href="https://github.com/hughcross" class="head">View on GitHub</a>
      <hr class="head"/>
      <div class="page">
  
  
            <h1 id="project_title">Hugh Cross' Jupyter Notebooks</h1>
          <h2 id="project_tagline">Links to my notebooks in html format</h2>
          <br/>
          <h2 id='teaching_notebooks'>Teaching Notebooks</h2>
          <p>The following notebooks are from a one day environmental DNA (eDNA) workshop, given as part of the Otago Carpentries Spring School. The eDNA workshop was team taught by Lara Urban, Gert-Jan Jeunen, and myself. I organised the workshop. All three contributed to content creation; the first section, <i>QC and Denoising</i> was taught by Gert-Jan; I covered the <i>Taxonomy Assignment</i> section; and Lara covered <i>Statistical Analysis</i>. I prepared all material for publication on the website. The material is also available on the <a href="https://otagocarpentries.github.io/bioinformatics-spring-school-2020/day5.html" target="_blank" rel="noopener noreferrer"><i>Spring School website</i></a> (Day 5). All of these lessons were taught using <a href="https://jupyter.org/hub" target="_blank" rel="noopener noreferrer">Jupyter Hub</a> on the <a href="https://www.nesi.org.nz/about-us/background" target="_blank" rel="noopener noreferrer">NeSI server</a>, which will now be the model for future workshops through Otago Carpentries.</p> 
          <p> <a href="https://hughcross.github.io/notebooks/introduction_qc_denoising_pub.html">Intro: QC and Denoising</a>
          <br/>
          <p> <a href="https://hughcross.github.io/notebooks/taxonomy_assign_pt1_pub.html">Taxonomy Assignment, Part 1</a>
          <br/>
          <p> <a href="https://hughcross.github.io/notebooks/taxonomy_assign_pt2_pub.html">Taxonomy Assignment, Part 2</a>
          <br/>
          <p> <a href="https://hughcross.github.io/notebooks/eDNA_statistical_analysis_notebook.html">Statistical Analysis</a>
          <br/>
          <h2 id='find_haps'>Finding Haplotypes in Alignments</h2>
          <p>The following three notebooks were used to create scripts for a project to determine frequencies of closely related haplotypes in a mock community experiment, in order to evaluate efficacy of denoising/clustering algorithms. The first two also server to illustrate my approach to script development. For larger scripts, my first step is usually to work out the different steps, using one or two example files or samples.     </p>

          <br/>
          <h2 id='convert_tables'>Converting tables</h2>
          <p> <a href="https://hughcross.github.io/notebooks/convert_obitab2qiime.html">Convert Obitools tab to Qiime2</a>
          <br/>
          <p> <a href="https://hughcross.github.io/notebooks/Create_genus_table_add_allLandPlants.html">Create sqlite DB from taxonomy table</a>
          <br/>
          <p> <a href="https://hughcross.github.io/notebooks/create_trnl_ref_from_blast.html">Create reference sequence from BLAST results</a>
          <br/>
          <p>
          <h2 id='gff_tools'>What the gff</h2>
          
          <p> <a href="https://hughcross.github.io/notebooks/change_values_in_column_of_gff.html">change_values_in_column_of_gff</a>
          <br/>
          <a href="https://hughcross.github.io/notebooks/convert_range_data_to_gff_with_diff_categories.html">convert_range_data_to_gff_with_diff_categories</a>
          <br/>
          <a href="https://hughcross.github.io/notebooks/convert_dN-gmap_to_dex-compat_map_gff_DEL.html">convert_dN-gmap_to_dex-compat_map_gff_DEL</a>
          <br/>
          <a href="https://hughcross.github.io/notebooks/finding_overlapping_ranges_between_two_gffs_dev.html">finding_overlapping_ranges_between_two_gffs_dev</a>



           </div>
</div>
          
          
